app/models/spree/product_question.rb

```diff
@@ -10,11 +10,19 @@ class Spree::ProductQuestion < ActiveRecord::Base
   scope :visible, -> { where(is_visible: true) }
   scope :answered, -> { joins(:product_answer) }
   scope :not_answered, -> { where.not(id: self.answered.pluck(:id)) }
-  scope :user_product_questions, ->(user_id) { where('spree_product_questions.user_id = ?', user_id) }
+  scope :user_product_questions, ->(user_id) { where('spree_product_questions.user_id = ?', user_id) }\
 
   validates :content, presence: true
 
   def answer_for_form
     self.product_answer.present? ? self.product_answer : self.build_product_answer
   end
+
+  def self.for_unauthorized_user
```

Comment:
subsequent to my comment above: https://github.com/spreehood/spree_products_qa/pull/22/files#r1615977168

your methods for authorized and unauthorized user would be consolidated to one function

```
def self.for_user(user = nil)
    if user
      visible.recent.left_outer_joins(:product_answer).where('spree_product_questions.user_id = ? OR spree_product_answers.id IS NOT NULL', user.id)
    else
      answered.visible.recent
    end
  end
  ```

I also added sorting by most recent, which is completely optional to do